### DISCUSSION TITLE
OpenAPI Paths

### DISCUSSION THREAD

**Discussion by atcol** [cue-lang/cue#1293] [Q&A] [answered]
Can I embed paths (URLs) for an openapi spec in my .cue?

Say I have this type (taken from another discussion on here):

```cue
#match: {
	id: int
	teams: [...string]
	score: [...int]
}
```
and I wish to expose this at `GET /api/v1/matches/{uid}`

**Comment by verdverm** [comment, marked as answer]
I have a hunch that this line means not yet: https://github.com/cue-lang/cue/blob/master/encoding/openapi/openapi.go#L213

**Reply by atcol** [reply]
For those wishing to use this for at least types, you can define your types in a .cue and your paths and other metadata in an openapi.yaml, and merge them with this:

```bash
cue export --out openapi openapi.cue | ./yq ea -P  'select(fileIndex == 0) * select(fileIndex == 1)' - openapi.yaml 
```

**Reply by verdverm** [reply]
You should also be able to do this with only the `cue` tool

```
cue export --out openapi types.cue | cue export --out yaml  - paths.yaml > openapi.yaml
```
Note, you can use `-` for stdin while specifying other files as well

I assume in either case, you have to take care in using `$ref` in the `paths` section to associate valid references

**Reply by atcol** [reply]
Ahh nice yes that's better. Thanks.

I'd still rather avoid manually linking up refs for schemas but that's Cue TOOD perhaps.
